swfiles/@spinw/powspec.m

```diff
@@ -213,9 +213,9 @@
 inpForm.defval = [inpForm.defval {false    false  0        'ebin'    'Sperp'    }];
 inpForm.size   = [inpForm.size   {[1 1]    [1 1]  [1 1]    [1 -4]     [1 -5]    }];
 
-inpForm.fname  = [inpForm.fname  {'fid'}];
-inpForm.defval = [inpForm.defval {-1   }];
-inpForm.size   = [inpForm.size   {[1 1]}];
+inpForm.fname  = [inpForm.fname  {'fid' , 'dE'}];
+inpForm.defval = [inpForm.defval {-1,      []}];
+inpForm.size   = [inpForm.size   {[1 1],   [-6, -7]}];
```

Comment:
Good spot, do you want `dE` to be hard or soft with the same default as `sw_egrid`? If hard, then I should probably set a default in `sw_fitpowder` or require it as a mandatory parameter in the init - which do you prefer?

Reply:
Ah I'd like `dE` to be a soft parameter like in `sw_egrid` so that the previous syntax where user doesn't specify `dE` as an argument to `powspec` would still work. Now I guess we can check and emit a warning that this might give them inconsistent results in certain edge cases and could make fitting harder but I think we should still allow the syntax without erroring.